paasta_tools/kubernetes_tools.py

```diff
@@ -1246,3 +1329,109 @@ def fromstring(cls, _str: str) -> int:
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_provider: BaseSecretProvider,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.create_namespaced_secret(
+        namespace="paasta",
+        body=V1Secret(
+            metadata=V1ObjectMeta(
+                name=f"paasta-secret-{service}-{secret}",
+                labels={"yelp.com/paasta_service": service},
+            ),
+            data={secret: base64.b64encode(secret_provider.decrypt_secret_raw(secret)).decode('utf-8')},
+        ),
+    )
+
+
+def update_secret(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_provider: BaseSecretProvider,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.replace_namespaced_secret(
+        name=f"paasta-secret-{service}-{secret}",
+        namespace="paasta",
+        body=V1Secret(
+            metadata=V1ObjectMeta(
+                name=f"paasta-secret-{service}-{secret}",
+                labels={"yelp.com/paasta_service": service},
+            ),
+            data={secret: base64.b64encode(secret_provider.decrypt_secret_raw(secret)).decode('utf-8')},
+        ),
+    )
+
+
+def get_kubernetes_secret_signature(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+) -> Optional[str]:
+    service = sanitise_service_name(service)
+    try:
+        signature = kube_client.core.read_namespaced_config_map(
+            name=f"paasta-secret-{service}-{secret}-signature",
+            namespace="paasta",
+        )
+    except ApiException as e:
+        if e.status == 404:
+            return None
+        else:
+            raise
+    if not signature:
+        return None
+    else:
+        return signature.data['signature']
+
+
+def update_kubernetes_secret_signature(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_signature: str,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.replace_namespaced_config_map(
+        name=f"paasta-secret-{service}-{secret}-signature",
+        namespace="paasta",
+        body=V1ConfigMap(
+            metadata=V1ObjectMeta(
+                name=f"paasta-secret-{service}-{secret}-signature",
+                labels={"yelp.com/paasta_service": service},
+            ),
+            data={'signature': secret_signature},
+        ),
+    )
+
+
+def create_kubernetes_secret_signature(
+    kube_client: KubeClient,
+    secret: str,
+    service: str,
+    secret_signature: str,
+) -> None:
+    service = sanitise_service_name(service)
+    kube_client.core.create_namespaced_config_map(
+        namespace="paasta",
+        body=V1ConfigMap(
+            metadata=V1ObjectMeta(
+                name=f"paasta-secret-{service}-{secret}-signature",
+                labels={"yelp.com/paasta_service": service},
+            ),
+            data={'signature': secret_signature},
+        ),
+    )
+
+
+def sanitise_service_name(
```

Comment:
surely we have this somewhere else already

Reply:
not exactly unless I've missed something. we have https://github.com/Yelp/paasta/blob/c44d689242df366d765cc734dfa5b083b4481a28/paasta_tools/marathon_tools.py#L974 for example

I think it's okay to live here, it is kinda k8s specific even if by coincidence we do something similar with marathon app names.